test/games/strategy/triplea/delegate/StratBombTest.java

```diff
@@ -0,0 +1,116 @@
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.Collections;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import games.strategy.engine.data.Change;
+import games.strategy.engine.data.changefactory.ChangeFactory;
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.ITestDelegateBridge;
+import games.strategy.engine.data.PlayerID;
+import games.strategy.engine.data.Route;
+import games.strategy.engine.data.RouteScripted;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.engine.random.IRandomSource;
+import games.strategy.engine.random.ScriptedRandomSource;
+import games.strategy.test.TestUtil;
+import games.strategy.triplea.Constants;
+import games.strategy.triplea.TripleAUnit;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.player.ITripleAPlayer;
+import games.strategy.triplea.xml.LoadGameUtil;
+
+public class StratBombTest {
+  private GameData m_data;
+
+  @Before
+  public void setUp() throws Exception {
+    m_data = LoadGameUtil.loadTestGame("ww2global40_2nd_edition_test.xml");
+  }
+
+  private ITestDelegateBridge getDelegateBridge(final PlayerID player) {
+    return GameDataTestUtil.getDelegateBridge(player, m_data);
+  }
+
+
+  @Test
+  public void testBombingRaid2targets() {
+    final Territory germany = m_data.getMap().getTerritory("Germany");
+    final Territory uk = m_data.getMap().getTerritory("United Kingdom");
+    final PlayerID germans = GameDataTestUtil.germans(m_data);
+    final PlayerID british = GameDataTestUtil.british(m_data);
+    // add a unit
+    final Unit stratBomber = GameDataTestUtil.bomber(m_data).create(british);
+    final Unit tacBomber1 = GameDataTestUtil.tacBomber(m_data).create(british);
+    final Unit tacBomber2 = GameDataTestUtil.tacBomber(m_data).create(british);
+    m_data.performChange(ChangeFactory.addUnits(uk, Collections.singleton(stratBomber)));
+    m_data.performChange(ChangeFactory.addUnits(uk, Collections.singleton(tacBomber1)));
+    m_data.performChange(ChangeFactory.addUnits(uk, Collections.singleton(tacBomber2)));
+    final BattleTracker tracker = new BattleTracker();
+    List<Unit> attackers = Collections.emptyList();
+    attackers.add(stratBomber);
+    attackers.add(tacBomber1);
+    attackers.add(tacBomber2);
+    HashMap<Unit, HashSet<Unit>> targets = null;
+   
+
+//    final Collection<Unit> enemyTargets = Match.getMatches(uk.getUnits().getMatches(new CompositeMatchAnd<>(Matches.enemyUnit(germans, m_data),              Matches.UnitIsAtMaxDamageOrNotCanBeDamaged(uk).invert(), Matches.unitIsBeingTransported().invert())), Matches.UnitIsLegalBombingTargetBy(stratBomber));
+
+    for( final Unit target : germany.getUnits().getUnits() ) {
```

Comment:
This for loop strikes me as a bit of a smell, namely too much logic in test code. If we have a test failure, it could take a good bit of effort to understand the test data setup, and why then we expect the behavior and return values that follow. 

This makes me think we really should have 3 different smaller test cases, one for each of the target types. The tests would then be much more focused and minimal. If desired, and if it woudl cover missing cases, a 4th test with mixed target types could then be added.